requirements.txt

```diff
@@ -25,3 +25,4 @@ PyYAML==4.2b2
 requests==2.20.0
 troposphere==2.3.3
 unittest2==1.1.0
+-e git+ssh://git@github.com/elifesciences/threadbare@develop#egg=threadbare
```

Comment:
(I am not supposed to look at this yet; but in the long term between the two routes of installing the library and pasting it into the `builder` project, the latter seems preferable as it makes fixes and refactorings much easier)

Reply:
you can totally look at it if you like ;) I'm quite proud of how well it's working.

I knew going in that separating it out into another project would troublesome, but the speed of development and isolation have been really good. Of course, now the two must be joined somehow that lets us make fixes quickly.

The `-e` flag means 'editable' and I've been working on it in `builder/venv/src/threadbare/`. Theoretically this should be good enough, but I suspect it won't be.

Another approach might be a sub-repository.

Copying and pasting threadbare into builder may be the least clever approach but probably the one with fewer problems. I'll investigate all three.

I'm able to completely launch and bootstrap an ec2 instance now with no problems. 

Exception handling has had some attention to make it consistent with Fabric but I need to take a second look, same with keyboard interrupts during parallel/multiprocessing operations.

Works in python2+3

Slices, dices.

Reply:
Went with a simple copy for now. Will sync changes back and forth manually until it's stable and gets a proper pypi release.

Reply:
> Slices, dices.

But does it make julienne?

Reply:
no. julienne requires python 4 and a mysterious fifth python parallel programming library that can only be detected when a thread within a subprocess attempts to run a geventlet in an asyncio future.